CustomerOrders/src/main/java/team/moxie/servlets/StaticFileServlet.java

```
@@ -0,0 +1,53 @@
+package team.moxie.servlets;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+
+public class StaticFileServlet extends HttpServlet {
+  
+  private String sourceFolder;
+  
+  public StaticFileServlet (String sourceFolder){
+    
+    this.sourceFolder = sourceFolder;
+  }
+  @Override protected void doGet (HttpServletRequest req, HttpServletResponse resp)
+  throws ServletException, IOException {
+    String url = req.getRequestURI();
+    System.out.println("GOt request for " + url);
+    OutputStream out = resp.getOutputStream();
+    try {
+      File file = Path.of(sourceFolder, url).toFile();
+      FileInputStream fis = new FileInputStream(file);
+      copyStream(fis, out);
+      
+    } catch (Exception e) {
+      try {
+        File fourOhFour = Path.of(sourceFolder, "/404.html").toFile();
+        FileInputStream fis = new FileInputStream(fourOhFour);
+        resp.setStatus(404);
+        copyStream(fis, out);
+        
+      }catch (Exception e2){
+      
+      }
```

Avoid empty catches, and also name the exceptions when you are nesting them like this.

---

I made this change